yarn-project/end-to-end/src/integration_archiver_l1_to_l2.test.ts

```diff
@@ -0,0 +1,171 @@
+import { NonNativeTokenContractAbi } from '@aztec/noir-contracts/examples';
+
+import { CircuitsWasm } from '@aztec/circuits.js';
+import { computeSecretMessageHash } from '@aztec/circuits.js/abis';
+import { DeployL1Contracts, deployL1Contract } from '@aztec/ethereum';
+import { Fr } from '@aztec/foundation/fields';
+import { DebugLogger } from '@aztec/foundation/log';
+import { PortalERC20Abi, PortalERC20Bytecode, TokenPortalAbi, TokenPortalBytecode } from '@aztec/l1-artifacts';
+import { Chain, HttpTransport, PublicClient, getContract } from 'viem';
+import { pointToPublicKey, setNextBlockTimestamp, setup } from './utils.js';
+import { Archiver } from '@aztec/archiver';
+
+describe('e2e_l1_to_l2_msg', () => {
+  let aztecNode: AztecNodeService;
+  let aztecRpcServer: AztecRPCServer;
+  let archiver: Archiver;
+  let accounts: AztecAddress[];
+  let logger: DebugLogger;
+  let config: AztecNodeConfig;
+
+  let contract: Contract;
+  let ethAccount: EthAddress;
+
+  let tokenPortalAddress: EthAddress;
+  let underlyingERC20Address: EthAddress;
+  let rollupRegistryAddress: EthAddress;
+  let tokenPortal: any;
+  let underlyingERC20: any;
+  let publicClient: PublicClient<HttpTransport, Chain>;
+
+  beforeEach(async () => {
+    let deployL1ContractsValues: DeployL1Contracts | undefined;
+    ({ aztecNode, aztecRpcServer, deployL1ContractsValues, accounts, config, logger } = await setup(2));
+    archiver = await Archiver.createAndSync(config);
+    rollupRegistryAddress = deployL1ContractsValues!.registryAddress;
+
+    const walletClient = deployL1ContractsValues.walletClient;
+    publicClient = deployL1ContractsValues.publicClient;
+
+    ethAccount = EthAddress.fromString((await walletClient.getAddresses())[0]);
+
+    // Deploy portal contracts
+    underlyingERC20Address = await deployL1Contract(walletClient, publicClient, PortalERC20Abi, PortalERC20Bytecode);
+    tokenPortalAddress = await deployL1Contract(walletClient, publicClient, TokenPortalAbi, TokenPortalBytecode);
+    underlyingERC20 = getContract({
+      address: underlyingERC20Address.toString(),
+      abi: PortalERC20Abi,
+      walletClient,
+      publicClient,
+    });
+    tokenPortal = getContract({
+      address: tokenPortalAddress.toString(),
+      abi: TokenPortalAbi,
+      walletClient,
+      publicClient,
+    });
+  }, 30_000);
+
+  afterEach(async () => {
+    await archiver.stop();
+    await aztecNode?.stop();
+    await aztecRpcServer?.stop();
+  });
+
+  const expectBalance = async (owner: AztecAddress, expectedBalance: bigint) => {
+    const ownerPublicKey = await aztecRpcServer.getAccountPublicKey(owner);
+    const [balance] = await contract.methods.getBalance(pointToPublicKey(ownerPublicKey)).view({ from: owner });
+    logger(`Account ${owner} balance: ${balance}`);
+    expect(balance).toBe(expectedBalance);
+  };
+
+  const deployContract = async (initialBalance = 0n, owner = { x: 0n, y: 0n }) => {
```

Comment:
alot of the boilerplate / functionality could be shared with the other test

Reply:
most of them require an insane number of params to be passed :( but I added some